django/urls/conf.py

```diff
@@ -70,7 +72,17 @@ def _path(route, view, kwargs=None, name=None, Pattern=None):
         pattern = Pattern(route, name=name, is_endpoint=True)
         return URLPattern(pattern, view, kwargs, name)
     else:
-        raise TypeError('view must be a callable or a list/tuple in the case of include().')
+        message = 'view must be a callable or a list/tuple in the case of include().'
+
+        # Detect the user passing `CBV()` instead of `CBV.as_view()`
+        if isinstance(view, View):
+            view_cls = view.__class__
+            message = (
+                "URL route '%s' should pass in '%s.as_view()' instead of '%s()'" %
+                (route, view_cls.__name__, view_cls.__name__)
+            )
```

Comment:
This error is raised when instantiating so we don't need to include a `route` in the message.